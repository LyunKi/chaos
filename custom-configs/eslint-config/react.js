module.exports = {
    extends: [
        'plugin:react/recommended',
        'airbnb',
        'airbnb/hooks',
        'plugin:@typescript-eslint/recommended',
        'prettier',
        'prettier/prettier',
        'prettier/react',
        'plugin:prettier/recommended',
    ],
    parser: '@typescript-eslint/parser',
    parserOptions: {
        ecmaFeatures: {
            jsx: true,
        },
        ecmaVersion: 12,
        sourceType: 'module',
    },
    plugins: ['react', '@typescript-eslint', 'react-hooks'],
    rules: {
        'no-shadow': 'off',
        '@typescript-eslint/no-shadow': ['error'],
    },
}
