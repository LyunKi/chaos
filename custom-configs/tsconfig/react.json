{
  "extends": "@tsconfig/create-react-app/tsconfig.json",
  "$schema": "https://json.schemastore.org/tsconfig"
}
