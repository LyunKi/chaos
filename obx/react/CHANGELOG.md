# @cloud-dragon/obx-react

## 0.8.1

### Patch Changes

- fix: fix components-rn exports config
- Updated dependencies
  - @cloud-dragon/react-utils@0.8.1
  - @cloud-dragon/obx-core@0.8.1

## 0.8.0

### Minor Changes

- feat: add components and ready to publish together

### Patch Changes

- Updated dependencies
  - @cloud-dragon/react-utils@0.8.0
  - @cloud-dragon/obx-core@0.8.0

## 0.7.0

### Minor Changes

- configure packages with same version

### Patch Changes

- Updated dependencies
  - @cloud-dragon/react-utils@0.7.0
  - @cloud-dragon/obx-core@0.7.0
