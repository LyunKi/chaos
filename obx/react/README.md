Reactive state management for react, based on @harmony/obx-core
