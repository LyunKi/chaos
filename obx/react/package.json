{
  "name": "@harmony/obx-react",
  "private": true,
  "version": "0.0.0",
  "files": [
    "dist",
    "index.d.ts"
  ],
  "main": "./dist/index.umd.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.mjs",
      "require": "./dist/index.umd.js"
    }
  },
  "scripts": {
    "build": "rimraf dist/ && rollup --config rollup.config.ts --configPlugin typescript && tsc -p tsconfig.lib.json",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:updateSnap": "jest --updateSnapshot"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "react": "^18.2.0",
    "@harmony/common-types": "workspace:*",
    "react-dom": "^18.2.0"
  },
  "dependencies": {
    "@harmony/obx-core": "workspace:*",
    "@harmony/react-utils": "workspace:*"
  }
}
