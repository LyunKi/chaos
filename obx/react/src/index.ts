export * from '@harmony/obx-core';
export * from './observer';
