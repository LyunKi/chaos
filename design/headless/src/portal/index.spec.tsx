import { render } from '@testing-library/react';
import { Portal } from './';

describe('Portal', () => {
  it('should render child in specify container', async () => {
    render(
      <Portal container={document.body}>
        <div id="remote">test</div>
      </Portal>
    );
    expect(document.body.lastChild).toMatchSnapshot();
  });
});
