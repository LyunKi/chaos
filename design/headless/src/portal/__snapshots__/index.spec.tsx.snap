// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal should render child in specify container 1`] = `
<div
  id="remote"
>
  test
</div>
`;
